### PR TITLE
New compatibility update for updates to CUDA

### PR DESCRIPTION
**Pull request by ChantalJuntao**
#16 

Also pushing project.toml up to a new version because the old version no longer seems compatible for GPU programming newer versions of Julia (I haven't tested which versions).

### CODE REVIEW

**Comment by ChantalJuntao** [comment]
I'm also not sure if these tests would run on CI?

File: Project.toml [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
 RegisterCore = "67712758-55e7-5c3c-8e85-dda1d7758434"
 RegisterMismatchCommon = "abb2e897-52bf-5d28-a379-6ca321e3b878"
 SharedArrays = "1a1011a3-84de-559e-8e89-a11a2f7dc383"
 
 [compat]
-CUDAdrv = "5, 6"
-CUDAnative = "2"
-CuArrays = "1"
-ImageCore = "0.8.1"
+ImageCore = "0.8.1, 0.9"
```
Comment (ChantalJuntao) [inline Project.toml:15]: CUDA = "3"

File: Project.toml [hunk 2]
```diff
@@ -1,26 +1,21 @@
...
 ImageCore = "a09fc81d-aa75-5fe9-8630-4744c3626534"
 Primes = "27ebfcd6-29c5-5fa9-bf4b-fb8fc14df3ae"
 RegisterCore = "67712758-55e7-5c3c-8e85-dda1d7758434"
 RegisterMismatchCommon = "abb2e897-52bf-5d28-a379-6ca321e3b878"
 SharedArrays = "1a1011a3-84de-559e-8e89-a11a2f7dc383"
 
 [compat]
-CUDAdrv = "5, 6"
-CUDAnative = "2"
-CuArrays = "1"
```
Comment (ChantalJuntao) [inline Project.toml:19]: Need to add CUDA = "3"

File: Project.toml [hunk 3]
```diff
@@ -1,26 +1,21 @@
 name = "RegisterMismatchCuda"
 uuid = "ee0d8d85-fa18-576c-9601-66ebb12862d9"
 authors = ["Tim Holy <tim.holy@gmail.com>"]
-version = "0.3.1"
+version = "0.4.1"
```
Comment (timholy) [inline Project.toml:4]:
```suggestion
version = "0.3.2"
```
This actually isn't a breaking change, in the sense that any user of this package doesn't have to change a single line of their code (I hope).

File: Project.toml (hunk 2 above)
Comment (timholy) [inline Project.toml:19]: CUDA = "3"

File: Project.toml [hunk 4]
```diff
@@ -1,26 +1,21 @@
...
 [compat]
-CUDAdrv = "5, 6"
-CUDAnative = "2"
-CuArrays = "1"
-ImageCore = "0.8.1"
+ImageCore = "0.8.1, 0.9"
 Primes = "0.4"
 RegisterCore = "0.2"
-RegisterMismatchCommon = "0.2.2"
+RegisterMismatchCommon = "0.2" #0.2.2
```
Comment (timholy) [inline Project.toml:18]:
```suggestion
RegisterMismatchCommon = "0.2.2"
```

File: src/RegisterMismatchCuda.jl [hunk 5]
```diff
@@ -116,8 +116,8 @@ CMStorage{T}(::UndefInitializer, aperture_width::NTuple{N,<:Real}, maxshift::Dim
 CMStorage{T}(::UndefInitializer, blocksize::NTuple{N,<:Real}, maxshift::Dims{N}; display=false) where {T<:Real,N} = CMStorage{T}(undef, blocksize, maxshift)
 CMStorage{T,N}(::UndefInitializer, blocksize::NTuple{N,<:Real}, maxshift::Dims{N}; display=false) where {T<:Real,N} = CMStorage{T,N}(undef, blocksize, maxshift)
 
-context(cms::CMStorage) = context(cms.num.C)
-context(a::CuArray) = a.ctx
+CUDA.context(cms::CMStorage) = context(cms.num.C)
+# context(a::CuArray) = a.ctx #deprecated
```
Comment (timholy) [inline src/RegisterMismatchCuda.jl:120]:
```suggestion
```

File: test/runtests.jl [hunk 6]
```diff
@@ -177,6 +177,7 @@ end
             for imsz in ((15,16), (14,17))
                 for maxshift in ((4,3), (3,2))
                     for gridsize in ((3,3), (2,1), (2,3), (2,2), (1,3))
+                        @show imsz, maxshift, gridsize
```
Comment (timholy) [inline test/runtests.jl:180]:
```suggestion
```

**Review by timholy — COMMENTED** [review]
Looks great, thanks for doing this! Minor suggestions but then we can merge.

**Comment by timholy** [comment]
Thanks for your contribution!

**Pull request merged**